data_steward/tools/generate_unioned_ehr_dataset.py

```diff
@@ -0,0 +1,293 @@
+
+
+def copy_mapping_tables(client, source_dataset, destination_dataset):
+    """
+    copies mapping tables from ehr_snapshot dataset to unioned_ehr_backup dataset
+    :param client: a BigQueryClient
+    :param source_dataset: the source dataset to copy from
+    :param destination_dataset: the destination dataset to copy to
+    """
+    for table in [
+            table.table_id
+            for table in client.list_tables(source_dataset)
+            if table.table_id.startswith(MAPPING_PREFIX)
+    ]:
+        source_table = f'{source_dataset}.{table}'
+        destination_table = f'{destination_dataset}.{table}'
+        if client.table_exists(source_table, source_dataset):
+            job = client.copy_table(source_table,
+                                    destination_table,
+                                    job_config=bigquery.job.CopyJobConfig(
+                                        write_disposition=bigquery.job.
+                                        WriteDisposition.WRITE_EMPTY))
+            job.result()
+            LOGGER.info(
+                f'Copied {source_table} to {destination_table} successfully')
+        else:
+            LOGGER.info(f'{source_table} does not exist')
+
+
+def copy_unioned_ehr_tables(client, source_dataset, destination_dataset):
+    """
+    copies unioned_ehr tables from ehr_snapshot dataset to unioned_ehr_backup dataset
+
+    :param client: a BigQueryClient
+    :param source_dataset: the source dataset to copy from
+    :param destination_dataset: the destination dataset to copy to
+    
+    """
+    for table in CDM_TABLES:
+        source_table_id = f'{source_dataset}.unioned_ehr_{table}'
+        source_table = f'{source_dataset}.{source_table_id}'
+        destination_table = f'{destination_dataset}.{table}'
+        if client.table_exists(source_table_id, source_dataset):
+            job = client.copy_table(source_table,
+                                    destination_table,
+                                    job_config=bigquery.job.CopyJobConfig(
+                                        write_disposition=bigquery.job.
+                                        WriteDisposition.WRITE_EMPTY))
+            job.result()
+            LOGGER.info(
+                f'Copied {source_table} to {destination_table} successfully')
+        else:
+            LOGGER.info(f'{source_table} does not exist')
+
+
+def main(raw_args=None):
+    args, kwargs = parse_unioned_ehr_args(raw_args)
+    pipeline_logging.configure(level=logging.INFO,
+                               add_console_handler=args.console_log)
+
+    impersonation_creds = auth.get_impersonation_credentials(
+        args.run_as_email, CDR_SCOPES)
+
+    client = BigQueryClient(args.project_id, credentials=impersonation_creds)
+
+    unioned_ehr_backup = create_dataset(client, args.release_tag, 'backup')
+
+    LOGGER.info('Creating destination CDM tables...')
+    create_cdm_tables(client, unioned_ehr_backup)
```

Comment:
This does not cover `aou_death`, since it's not a part of `CDM_TABLES`.

Reply:
I think we still need this.